source/Calamari/Kubernetes/ResourceStatus/ResourceStatusCheckTask.cs

```diff
@@ -43,10 +43,10 @@ public class ResourceStatusCheckTask
 
             var definedResources = resources.ToArray();
             var checkCount = 0;
+            var result = new Result();
```

Comment:
Moved this outside the loop so it is consistent with the implementation before the glob path changes:
https://github.com/OctopusDeploy/Calamari/blob/b90549801bfc8d6f9e14ef6ee9f00a7fa80f2a1c/source/Calamari/Kubernetes/ResourceStatus/ResourceStatusChecker.cs#L47

This makes sure we send back the updated resources comparing with the last run, instead of an empty list of resources.

@scme0 Can you double check on this?

Reply:
Actually, moving this outside will cause it to be shared by threads and is therefore not safe. Sorry I misunderstood this part at the beginning, I'm reverting this.